src/Compilers/CSharp/Portable/Lowering/Instrumentation/Instrumenter.cs

```diff
@@ -242,12 +242,18 @@ public virtual BoundStatement InstrumentSwitchStatement(BoundSwitchStatement ori
             return InstrumentStatement(original, rewritten);
         }
 
-        public virtual BoundStatement InstrumentBoundPatternSwitchStatement(BoundPatternSwitchStatement original, BoundStatement rewritten)
+        public virtual BoundStatement InstrumentPatternSwitchStatement(BoundPatternSwitchStatement original, BoundStatement rewritten)
         {
             Debug.Assert(original.Syntax.Kind() == SyntaxKind.SwitchStatement);
             return InstrumentStatement(original, rewritten);
         }
 
+        public virtual BoundStatement InstrumentPatternSwitchWhenClause(BoundExpression original, BoundStatement rewritten)
```

Comment:
>public virtual BoundStatement InstrumentPatternSwitchWhenClause(BoundExpression original, BoundStatement rewritten) [](start = 8, length = 115)

I think we should be more explicit about the contract both in the name of the method and in the name of parameters, similar to InstrumentDoStatementConditionalGotoStart, for example. Rewritten is a conditional goto and knowledge about that will be very helpful to implementers because any statement placed after this one will be executed conditionally, not always. Understanding the nature of the condition and where does the statement jump to could be helpful too.